packages/react-native/Libraries/Text/TextInput/Singleline/RCTUITextField.mm

```diff
@@ -27,11 +29,33 @@ - (instancetype)initWithFrame:(CGRect)frame
 
     _textInputDelegateAdapter = [[RCTBackedTextFieldDelegateAdapter alloc] initWithTextField:self];
     _scrollEnabled = YES;
+    _disablePlaceholderScaling = NO;
   }
 
   return self;
 }
 
+- (void)didAddSubview:(UIView *)subview
+{
+    [super didAddSubview:subview];
+    
+     if ([subview isKindOfClass:UILabel.class]) {
+         _placeholderLabel = (UILabel *)subview;
+         _placeholderLabel.adjustsFontSizeToFitWidth = !_disablePlaceholderScaling;
+     }
+}
+
+- (void)updateDisablePlaceholderScaling:(BOOL)val
+{
+    _disablePlaceholderScaling = val;
+    
+    if (_placeholderLabel != nil) {
+        _placeholderLabel.adjustsFontSizeToFitWidth = !_disablePlaceholderScaling;
+        [_placeholderLabel setNeedsDisplay];
+    }
```

Comment:
I wonder whether iterating over subviews and checking fo `UILabel` here wouldn't be a better approach. Realistically, the label should be the only child so it would be one iteration and we avoid overriding `didAddSubview` and keeping a reference to it.

I believe it should already be there at the point where props are set, though it would be best to check that.